pkg/util/pki/parse.go

```diff
@@ -252,8 +252,14 @@ func (c *chainNode) toBundleAndCA() (PEMBundle, error) {
 
 	for {
 		// If the issuer is nil, we have hit the root of the chain. Assign the CA
-		// to this certificate and stop traversing.
+		// to this certificate and stop traversing. If the root certificate is not
+		// a self-signed certificate, i.e. is not a root CA, append the certificate
+		// to the certificate chain.
+
 		if c.issuer == nil {
+			if !isSelfSignedCertificate(c.cert) {
```

Comment:
[This comment] is still valid. We need to be checking that at least 2 certs were parsed if we are to copy.

```suggestion
			if len(certs) > 1 && !isSelfSignedCertificate(c.cert) {
```

This test case should be asserted IMO

```golang
		"if only a single certificate was parses, CAPEM should remain empty": {
			inputBundle:  joinPEM(leaf.pem),
			expPEMBundle: PEMBundle{ChainPEM: joinPEM(leaf.pem), CAPEM: nil},
			expErr:       false,
		},
```

@SgtCoDFish

Reply:
> This comment is still valid.

Agreed about the first paragraph of that comment specifically, yeah.

> This test case should be asserted IMO

I think that's a good test case to add, yeah. There're related test cases too, which might be worth adding for completeness here:

```go
		"if only a single intermediate certificate was parsed, ChainPEM should contain a single certificate and CAPEM should remain empty": {
			inputBundle:  joinPEM(intA1.pem),
			expPEMBundle: PEMBundle{ChainPEM: joinPEM(intA1.pem), CAPEM: nil},
			expErr:       false,
		},
		"if only a single root certificate was parsed, ChainPEM should contain a single certificate and CAPEM should remain empty": {
			inputBundle:  joinPEM(root.pem),
			expPEMBundle: PEMBundle{ChainPEM: joinPEM(root.pem), CAPEM: nil},
			expErr:       false,
		},
```

Reply:
Not sure, I think you are right- we do expect that to be present in `CAPEM`.